pennylane/devices/legacy_facade.py

```diff
@@ -0,0 +1,428 @@
+
+        if execution_config is None or execution_config.gradient_method == "best":
+            validation_methods = (
+                self._validate_backprop_method,
+                self._validate_adjoint_method,
+                self._validate_device_method,
+            )
+            return any(validate(circuit) for validate in validation_methods)
+
+        if execution_config.gradient_method == "backprop":
+            return self._validate_backprop_method(circuit)
+        if execution_config.gradient_method == "adjoint":
+            return self._validate_adjoint_method(circuit)
+        if execution_config.gradient_method == "device":
+            return self._validate_device_method(circuit)
+        return False
+
+    # pylint: disable=protected-access
+    def _create_temp_device(self, batch):
+        """Create a temporary device for use in a backprop execution."""
+        params = []
+        for t in batch:
+            params.extend(t.get_parameters(trainable_only=False))
+        interface = qml.math.get_interface(*params)
+        if interface == "numpy":
+            return self._device
+
+        mapped_interface = qml.workflow.execution.INTERFACE_MAP.get(interface, interface)
+
+        backprop_interface = self._device.capabilities().get("passthru_interface", None)
+        if mapped_interface == backprop_interface:
+            return self._device
+
+        backprop_devices = self._device.capabilities().get("passthru_devices", None)
+
+        if backprop_devices is None:
+            raise qml.DeviceError(f"Device {self} does not support backpropagation.")
+
+        if backprop_devices[mapped_interface] == self._device.short_name:
+            return self._device
+
+        # create new backprop device
+        expand_fn = self._device.expand_fn
+        batch_transform = self._device.batch_transform
+        if hasattr(self._device, "_debugger"):
+            debugger = self._device._debugger
+        else:
+            debugger = "No debugger"
+        tracker = self._device.tracker
+
+        backprop_devices = self._device.capabilities().get("passthru_devices", None)
+
+        new_device = qml.device(
+            backprop_devices[mapped_interface], wires=self._device.wires, shots=self._device.shots
+        )
+        new_device.expand_fn = expand_fn
+        new_device.batch_transform = batch_transform
+        if debugger != "No debugger":
+            new_device._debugger = debugger
+        new_device.tracker = tracker
+
+        return new_device
+
+    # pylint: disable=protected-access
+    def _update_original_device(self, temp_device):
+        """After performing an execution with a backprop device, update the state of the original device."""
+        # Update for state vector simulators that have the _pre_rotated_state attribute
+        if hasattr(self._device, "_pre_rotated_state"):
+            self._device._pre_rotated_state = temp_device._pre_rotated_state
```

Comment:
Won't this introduce interface data into the original device?

Reply:
Just copying code from the qnode.  Assuming it was in the qnode for some reason.